src/cli/peakdetector/PeakDetectorCLI.cpp

```diff
@@ -627,57 +671,92 @@ void PeakDetectorCLI::writeReport(string setName,QString jsPath,QString nodePath
+			bool valid_sample_cohort = false;
 			
-			// jspath and nodepath are very important here..node executable will be used to connect to polly, with the help of index.js script..
-			QString upload_project_id = UploadToPolly(jsPath,nodePath,QStringList()<<csv_filename+".csv"<<json_filename+".json"<<compound_DB_filename+".csv",creds); //add more files to upload, if desired..
-			if (upload_project_id!=""){ //That means the upload was successfull, in that case, redirect the user to polly..
-				QString redirection_url = QString("<a href='https://polly.elucidata.io/main#project=%1&auto-redirect=firstview'>Go To Polly</a>").arg(upload_project_id);
-				qDebug()<<"redirection url - \n"<<redirection_url;
-				QString filename=writable_temp_dir+QDir::separator()+datetimestamp+"_redirection_url.txt";
-				QFile file( filename );// redirection url will be written to a file..In future it will be replaced by email feature..
-				if ( file.open(QIODevice::ReadWrite) )
-				{
-					QTextStream stream( &file );
+			if (!sample_cohort_file.isEmpty()) {
+				valid_sample_cohort = validSampleCohort(sample_cohort_file,loadedSamples);
+			}
+
+			if (valid_sample_cohort) {
+				bool status_sample_copy = QFile::copy(sample_cohort_file, 
+												sample_cohort_filename+".csv");
+				qDebug() << "sample cohort copy status - " << status_sample_copy;
+				redirect_to = "relative_lcms_elmaven";
+			}
+			else {
+				qDebug() << "There was some problem with the sample cohort file, you will be redirected to an interface on polly where you can make the cohort file again";
+				makeSampleCohortFile(sample_cohort_filename+".csv",loadedSamples);
+			}
+			
+			//add more files to upload, if desired..
+			QStringList files_to_be_uploaded = QStringList() << csv_filename + ".csv" << 
+													json_filename + ".json" << 
+													compound_DB_filename + ".csv" <<
+													sample_cohort_filename+".csv";
+
+			//jspath and nodepath are very important here..node executable will be used to connect to polly, with the help of index.js script..
+			QString upload_project_id = UploadToPolly(jsPath, nodePath, files_to_be_uploaded, 
+											creds, pollyProject); 
+			
+			//That means the upload was successful, redirect the user to Polly..
+			if (!upload_project_id.isEmpty()) {
+				QString redirection_url = QString("<a href='https://polly.elucidata.io/main#project=%1&auto-redirect=%2&elmavenTimestamp=%3'>Go To Polly</a>").arg(upload_project_id).arg(redirect_to).arg(datetimestamp);
+				qDebug() << "redirection url - \n" << redirection_url;
+				QString filename = QStandardPaths::writableLocation(
+									QStandardPaths::QStandardPaths::GenericConfigLocation) + 
+									QDir::separator() + 
+									datetimestamp + 
+									"_redirection_url.txt";
+				
+				qDebug() << "writing url to this file -" << filename;
+				//redirection url will be written to a file..In future it will be replaced by email feature..
```

Comment:
@Ky5595 Email feature has been added. Can we remove the file now?

Reply:
@shubhra-agrawal 
In case users don't get an email, we still want to have some source from where they can be redirected to polly.
So, I believe we should retain this functionality.